Don't substitute in symbolic shapes to shape compute graph

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59814 Test shape analysis with opinfos
* #59812 add ssupport for a few more opinfos in jit
* **#59811 Don't substitute in symbolic shapes to shape compute graph**
* #59810 small cleanups
* #59809 Only optimize after change (redo)

We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

Differential Revision: [D30200059](https://our.internmc.facebook.com/intern/diff/D30200059)

Commit:
Don't substitute in symbolic shapes to shape compute graph

[ghstack-poisoned]

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/symbolic_shape_analysis.cpp:
```diff
@@ -166,12 +163,12 @@ struct SymbolicShapeAnalyzer {
+    // which resolve to symbolic dimension values with their concrete symbolic
+    // shape value. Because symbolic dimensions are represented as negative numbers and
+    // are not real values, inserting them as constants in the graph would invalidate
+    // the graph for further use. Instead, we keep track of what there value would be
```
typo: s/there/their/

ZolotukhinM commented on torch/csrc/jit/passes/symbolic_shape_analysis.cpp:
```diff
@@ -198,10 +195,12 @@ struct SymbolicShapeAnalyzer {
             }
-            if (tensor_shape[*norm_index].is_static() ||
-                substitute_symbolic_dims) {
+            if (tensor_shape[*norm_index].is_static()) {
```
It's a comment on already existing code, but still: is my understanding correct that `node_` here corresponds to a node in the real graph on which we run the shape prop pass, while the `graph_` is the shape-compute graph? If so, can we reflect in their name that they are explicitly separate? It would help to comprehend the code I think. Otherwise I keep catching myself assuming that `graph_` must be the same as `node_->owningGraph()`, which I think is wrong.

eellison replied:
yes, that is correct. what would you suggest, naming-wise?

ZolotukhinM replied:
Maybe explicitly call it "shape_function_graph_" or something like that?

ZolotukhinM reviewed (approved):
Looks good to me!

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

[ghstack-poisoned]

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

[ghstack-poisoned]

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

[ghstack-poisoned]

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30200059).

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

Differential Revision: [D30200059](https://our.internmc.facebook.com/intern/diff/D30200059)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30200059).

Commit:
Update on "Don't substitute in symbolic shapes to shape compute graph"


We don't want to actually substitute in symbolic shapes, because it invalidates the partially evaluated graph for further use.

Differential Revision: [D30200059](https://our.internmc.facebook.com/intern/diff/D30200059)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30200059).
